### PR TITLE
Fix/tools

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR performs multiple fixes on the tools as a result of the recent failures.

### CODE REVIEW

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py [hunk 1]
```diff
@@ -681,114 +681,117 @@ def extract_question(prompt: str) -> str:
...
+                    input_tokens=response_reasoning.usage.prompt_tokens
+                    + response.usage.prompt_tokens,
+                    output_tokens=response_reasoning.usage.completion_tokens
+                    + response.usage.completion_tokens,
+                    model=engine,
+                    token_counter=count_tokens,
+                )
+            return results, reasoning_prompt + "////" + prediction_prompt, None, counter_callback
+    except Exception as e:
+        return f"Invalid response. The following issue was encountered: {str(e)}", "", None, None
```
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:797]: check response here

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py [hunk 2]
```diff
@@ -681,114 +681,117 @@ def extract_question(prompt: str) -> str:
...
-            engine=engine,
-            google_api_key=google_api_key,
-            google_engine_id=google_engine_id,
-            counter_callback=counter_callback,
-            source_links=kwargs.get("source_links", None),
-            num_urls=num_urls,
-            temperature=temperature,
-            max_tokens=max_tokens,
-        )
+    try:
```
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:684]: @richardblythman @moarshy Wrapping in a try/except like this really helps to debug when things go wrong. I suggest we take this approach going forward. If you have some cleaner way of doing it than with simple try/except like this please propose.

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py (hunk 1 above)
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:797]: Check response here

File: packages/nickcom007/customs/prediction_request_sme/prediction_request_sme.py [hunk 3]
```diff
@@ -482,6 +482,7 @@ def run(**kwargs) -> Tuple[str, Optional[str], Optional[Dict[str, Any]], Any]:
                 "Moderation flagged the prompt as in violation of terms.",
                 prediction_prompt,
                 None,
+                counter_callback,
```
Comment (0xArdi) [inline packages/nickcom007/customs/prediction_request_sme/prediction_request_sme.py:485]: was missing on this branch of exec. lead to failure

File: packages/valory/agents/mech/aea-config.yaml [hunk 4]
```diff
@@ -117,6 +119,10 @@ dependencies:
     version: ==0.4.13
   markdownify:
     version: ==0.11.6
+  moviepy:
+    version: ==1.0.3
+  replicate:
+    version: ==0.15.7
```
Comment (0xArdi) [inline packages/valory/agents/mech/aea-config.yaml:125]: required by the shorts tool

File: pyproject.toml [hunk 5]
```diff
@@ -68,4 +68,4 @@ readability-lxml = "0.8.1"
 docstring-parser = "0.15"
 faiss-cpu = "1.7.4"
 pypdf2 = "^3.0.1"
-lxml = {extras = ["html-clean"], version = "^5.2.0"}
+lxml = {extras = ["html-clean"], version = "^5.1.0"}
```
Comment (0xArdi) [inline pyproject.toml:71]: 5.2.0 caused issues, moved to 5.1.0 seems to be stable

File: tox.ini [hunk 6]
```diff
@@ -64,7 +64,7 @@ deps =
     pandas==2.1.1
     tiktoken==0.5.1
     python-dateutil==2.8.2
-    lxml==5.2.0
+    lxml==5.1.0
```
Comment (0xArdi) [inline tox.ini:67]: same

File: packages/napthaai/customs/prediction_request_rag_claude/component.yaml
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_rag_claude/component.yaml:1]: component.yaml files MUST be added

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py (hunk 2 above)
Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:684]: OK sure, sounds good.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**